packages/xforms-engine/src/lib/NumberParsers.ts

```diff
@@ -0,0 +1,25 @@
+export const parseToInteger = (value: string | null): number | null => {
+	if (value === null) {
+		return null;
+	}
+
+	const parsed = Number(value);
+	if (typeof value !== 'string' || value.trim() === '' || !Number.isInteger(parsed)) {
+		throw new Error(`Expected an integer, but got: ${value}`);
+	}
+
+	return parsed;
+};
+
+export const parseToFloat = (value: string | null): number | null => {
+	if (value === null) {
+		return null;
+	}
+
+	const parsed = Number(value);
+	if (typeof value !== 'string' || value.trim() === '' || isNaN(parsed)) {
```

Comment:
Up to you if you want to change this to
```suggestion
	if (typeof value !== 'string' || value.trim() === '' || !Number.isFinite(parsed)) {
```

Note: in general I think for _just_ `NaN` checking, `Number.isNaN` is preferred over the global `isNaN`. But if you don't want to do the suggested change I'd just leave as-is. This is really something we should have a lint rule for.